### DISCUSSION TITLE
Lume for SSG in Fresh

### DISCUSSION THREAD

**Discussion by erlend-sh** [lumeland/lume#236] [Ideas]
I’m a bit out of my depth there, but could Lume possibly be used to add support for static site generation in the Fresh frontend framework?

Apparently SSG is not currently possible in Fresh, and it’s not straightforward to add it: https://github.com/denoland/fresh/discussions/254

**Comment by oscarotero** [comment]
If you mean creating a site combining Fresh and Lume, I don't see how it can be done.

Lume has it's own way to generate the static files (similar to 11ty, jekyll or Hugo) that is a bit different from Fresh. It has support for JSX (React as primary library but you can use also Preact), so if your site is composed only with static pages, I think it should be easy to port to Lume. But Lume hasn't a built in concept of "islands" to create interactive components in the client side magically, so this part may take a bit more work (or maybe not, not sure).
